### PR TITLE
Question deps

### PR DESCRIPTION
**Pull request by thelgevold**
This is not a final solution, but I am experimenting with RXJS/Redux flows in our application. 

Will have to iterate on this, but it's at least a start.

### CODE REVIEW

File: src/core/components/select-field/select-field.ts [hunk 1]
```diff
@@ -1,24 +1,39 @@
...
+  }
 
+  ngOnInit(){
+      
+      if(!this.viewModel.items){
+        this._codeListService.getCodeList(this.viewModel.href,this.viewModel.id);
+      }
+      else{
+          this.items = this.viewModel.items;
+      }
```
Comment (thelgevold) [inline src/core/components/select-field/select-field.ts]: Can probably get rid of items and use directly from viewModel.items

File: src/core/services/code-list-service.spec.ts [hunk 2]
```diff
@@ -46,9 +46,9 @@ describe('CodeListService Service', () => {
       connection.mockRespond(new Response(new ResponseOptions({body: sampleCodeList})));
     });
 
-    service.getCodeList('testHref').subscribe((data) => {
-      expect(data).toBe(sampleCodeList.listItems);
-    });
+    // service.getCodeList('testHref','id').subscribe((data) => {
+    //   expect(data).toBe(sampleCodeList.listItems);
+    // });
```
Comment (thelgevold) [inline src/core/services/code-list-service.spec.ts]: Will bring back the test when this stabilizes

File: src/assets/meta/json-schema-v3/dependencies.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      "parent":"Country",
+      "optional": true,
+      "hidden": false,
+      "shortLabelName": "Country",
+      "longLabelName": "Country",
+      "codeList": {
+        "links": [
+          {
+            "rel": "codelist",
+            "href": "./assets/codelists/states-usa.json",
```
Comment (jpdonaghy) [inline src/assets/meta/json-schema-v3/dependencies.json]: A dependency would be defined on the href like "./assets/codelists/country-subdivisions?countryCode={#countryCode}"

File: src/assets/meta/json-schema-v3/dependencies.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{
+  "properties" : [
+    {
```
Comment (jpdonaghy) [inline src/assets/meta/json-schema-v3/dependencies.json]: All of these properties need to be named like : 

"address/countryCode" : { metadata }

This name is a fully qualified path into the model.  So the model binding for this field would be : 

{
   address : {
      countryCode: 'US'
   }
}

**Comment by jpdonaghy** [comment]
This is going in the right direction.  I think before we go any further we have to get our model bindings correct.  This requires us to name the meta property objects.  We then must add a 'path' property to the ViewModelBase so the value can be bound to the form model properly.  The meta defines the structure of the POST body this way.

**Comment by jpdonaghy** [comment]
Go ahead and merge this and I'll work on the model bindings in parallel.

File: src/assets/meta/json-schema-v3/dependencies.json (hunk 4 above)
Comment (thelgevold) [inline src/assets/meta/json-schema-v3/dependencies.json]: Yeah, that pattern should make it possible to build the tree and correlate dependencies.

**Pull request merged**